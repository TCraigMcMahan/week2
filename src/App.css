.App {
  text-align: center;
}

.App-header {
  background-color: white;
  height: 50px;
  padding: 10px;
  border-top: 0px;
  border-left: 0px;
  border-right: 0px;
  border-bottom: 2px;
  border-bottom-color: lightgray;
  border-style: solid;
  margin-left: 10px;
  margin-bottom: 25px;
  width: 1075px;
  max-width: 1100px;
}

.App-title {
  font-size: 1.5em;
  text-align: left;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.App-card-container {
  display: flex;
  flex-wrap: wrap;
  height: 600px;
  max-height: 600px;
  width: 1100px;
  max-width: 1100px;
  align-items: center;
  background-color: white;
}

.App-card-container::before {
  content: " ";
  position: absolute;
  z-index: -1;
  top: 50px;
  left: 8px;
  right: 20px;
  width: 95px;
  border-bottom: 1.5px solid black;
}

.App-missed-card-container {
  display: flex;
  flex-wrap: wrap;
  height: 350px;
  max-height: 350px;
  width: 1100px;
  max-width: 1100px;
  align-items: center;
  background-color: white;
}

.App-missed-card-container::before {
  content: " ";
  position: absolute;
  z-index: -1;
  top: 725px;
  left: 8px;
  right: 20px;
  width: 240px;
  border-bottom: 1.5px solid black;
}
  