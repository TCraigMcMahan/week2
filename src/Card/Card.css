* {
    box-sizing: border-box;
}

.side {
    flex: 30%;
    padding: 10px;
}

.main {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    flex: 70%;
    padding: 10px;
}

.row {
    height: 275px;
    max-height: 275px;
    width: 525px;
    max-width: 525px;
    background-color: white;
    border: lightgray;
    border-width: 2px;
    border-style: solid;
    margin: 10px;
}

.Card-img {
    float: left;
    width: 250px;
    height: 250px;
    padding: 10px;
}

.Title {
    box-align: left;    
}

.Author-img {
    width: 50px;
    height: 50px;
    border-radius: 50%;
    padding: 3px;
    background-size: auto;
}

.authorInfo {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: row;
    height: 75px;
    max-height: 75px;
    font-size: 12px;
}


.authorInfo-namedate {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    padding: 10px;
    height: 75px;
    max-height: 75px;
    width: 200px;
    font-size: 15px;
}

.titleStyle {
    font-weight: bold;
    text-align: left;    
}

.bookmarkStyle {
    width: 50%;
    height: 50%;
}

.authorNameStyle {
    font-weight: bold;
    text-align: left;
}

.authorMemberContainer {
    height: 50px;
    width: 50px;
    margin-left: 5px;
    background-image: url('../img/arcs.png');
}

.upperScreen {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    height: 150px;
    max-height: 150px;
    font-size: 15px;
}

.bookmarkContainer {
    height: 25px;
    width: 25px;  
    padding-top: 10px;
    margin-left: 10px;
}

.bookmarkContainer img {
    height: auto;
    width: 90%;
}