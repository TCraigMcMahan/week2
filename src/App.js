import React, { Component } from 'react';
import './App.css';
import Card from './Card/Card.js';
import MissedCard from './MissedCard/MissedCard.js';

import articles from './your-articles.json';
import missedArticles from './missed-articles.json';

class App extends Component {
  render() {

    return (
      <div className="App">
        <div className="App-header">
          <div className="App-title">For you</div>
        </div >
        <div className="App-card-container" >
          {articles.map((card, idx) => {
            return <Card key={idx} card={card} />;
          })}
        </div >

        <div className="App-header">
          <div className="App-title">In case you missed it</div>
        </div >
        <div className="App-missed-card-container" >
          {missedArticles.map((card, idx) => {
            return <MissedCard key={idx} missedcard={card} />;
          })}
        </div >
      </div>
    );
  }
}

export default App;
