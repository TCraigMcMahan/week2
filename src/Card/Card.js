import React, { Component } from 'react';
import './Card.css';
import Audio from '../Icons/Audio';
import '../Icons/Audio.css';
import speaker from '../img/icon-speaker.png';
import star from '../img/icon-star.svg';
import bookmark from '../img/icon-bookmark.svg';
import PropTypes from 'prop-types';

class Card extends Component {
    render() {

        const {
            title,
            image,
            description,
            link,
            author,
            postedDate,
            minutesToRead,
            hasAudioAvailable,
            memberPreview
        } = this.props.card;


        const showMember = (props) => {
            if (author.isMediumMember) {
                return (
                    <div className="authorMemberContainer">
                        <img className="Author-img" src={author.image} alt="" />
                    </div>
                )
            }
            else {
                return (
                    <div>
                        <img className="Author-img" src={author.image} alt="" />
                    </div>
                )
            }
        }

        const getDate = (props) => {
            const monthNum = postedDate.toString().substring(5, 7);
            let monthAbbrev = '';
            const dayNum = postedDate.toString().substring(8, 10);
            let modDayNum = dayNum;
            let dateStr = '';

            switch (parseInt(monthNum, 10)) {
                case 1:
                    monthAbbrev = 'Jan';
                    break;
                case 2:
                    monthAbbrev = 'Feb';
                    break;
                case 3:
                    monthAbbrev = 'Mar';
                    break;
                case 4:
                    monthAbbrev = 'Apr';
                    break;
                case 5:
                    monthAbbrev = 'May';
                    break;
                case 6:
                    monthAbbrev = 'Jun';
                    break;
                case 7:
                    monthAbbrev = 'Jul';
                    break;
                case 8:
                    monthAbbrev = 'Aug';
                    break;
                case 9:
                    monthAbbrev = 'Sep';
                    break;
                case 10:
                    monthAbbrev = 'Oct';
                    break;
                case 11:
                    monthAbbrev = 'Nov';
                    break;
                case 12:
                    monthAbbrev = 'Dec';
                    break;
                default:
                    monthAbbrev = '';
            }

            if (dayNum[0] === '0') {
                modDayNum = dayNum[1];
            }
            else {
                modDayNum = dayNum;
            }

            dateStr = monthAbbrev + ' ' + modDayNum;

            return dateStr;
        }

        return (

            <div className="row">
                <div className="side" >
                    <img className="Card-img" src={image} alt="" />
                </div>
                <div className="main">
                    <div className="upperScreen">
                        <Audio hasAudioAvailable={hasAudioAvailable} memberPreview={memberPreview} />
                        <div className="titleStyle"> {title} </div>
                        <br></br>
                        <div align="left">{description}</div>
                    </div>
                    <div className="authorInfo" >
                        <div>
                            {showMember(author.isMediumMember)}
                        </div>
                        <div className="authorInfo-namedate">
                            <div className="authorNameStyle">{author.name}</div>
                            <div align="left"> {getDate()} - {minutesToRead} min read</div>
                        </div>
                        <div className="bookmarkContainer">
                        <img className="bookmarkStyle" src={bookmark} alt="Bookmark" />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

Card.propTypes = {
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    description: PropTypes.string,
    author: PropTypes.instanceOf(Object),
    postedDate: PropTypes.string.isRequired,
    minutesToRead: PropTypes.number,
    hasAudioAvailable: PropTypes.bool.isRequired,
    memberPreview: PropTypes.bool.isRequired
};

Card.defaultProps = {
    name: 'Anonymous',
    memberPreview: false,
}

export default Card;
