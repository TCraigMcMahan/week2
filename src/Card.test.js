import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import './App.css';
import Card from './Card/Card.js';
import './Card/Card.css';
import testCard from './cardTest.json';









describe('Card', () => {


    it('renders without crashing with data', () => {
        const author = {
            name: "Kris Gage",
            image: "https://cdn-images-1.medium.com/fit/c/80/80/1*e7o4kbk8ofT6eKaLrO0jaw.png",
            isMediumMember: false
        };
        const div = document.createElement('div');
        const element = testCard.map((card, index) => {
            return <Card key={index} card={card} title="Do Something You Can't Win At"
            image="https://cdn-images-1.medium.com/max/800/1*cioT9CBBV2s0svet9LqKZA.png"
            postedDate="2018-03-30T07:00:00.000Z"
            author={author}
            hasAudioAvailable={false}  />
        });
        ReactDOM.render(element, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with title, description, image, etc.', () => {
        const element = testCard.map((card, index) => {
            return <Card key={index} card={card} title="test" />
        });
        const component = TestRenderer.create(element);
        expect(component.toJSON()).toMatchSnapshot();
    });
});    
