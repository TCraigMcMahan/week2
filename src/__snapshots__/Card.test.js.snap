// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with title, description, image, etc. 1`] = `
<div
  className="row"
>
  <div
    className="side"
  >
    <img
      alt=""
      className="Card-img"
      src="https://cdn-images-1.medium.com/max/800/1*cioT9CBBV2s0svet9LqKZA.png"
    />
  </div>
  <div
    className="main"
  >
    <div
      className="upperScreen"
    >
      <div
        className="titleStyle"
      >
         
        Do Something You Can't Win At
         
      </div>
      <br />
      <div
        align="left"
      >
        And make it something you love
      </div>
    </div>
    <div
      className="authorInfo"
    >
      <div>
        <div>
          <img
            alt=""
            className="Author-img"
            src="https://cdn-images-1.medium.com/fit/c/80/80/1*e7o4kbk8ofT6eKaLrO0jaw.png"
          />
        </div>
      </div>
      <div
        className="authorInfo-namedate"
      >
        <div
          className="authorNameStyle"
        >
          Kris Gage
        </div>
        <div
          align="left"
        >
           
          Mar 30
           - 
          3
           min read
        </div>
      </div>
      <div
        className="bookmarkContainer"
      >
        <img
          alt="Bookmark"
          className="bookmarkStyle"
          src="icon-bookmark.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
